src/state/slices/opportunitiesSlice/resolvers/uniV2/index.ts

```diff
@@ -1,159 +1,237 @@
-        underlyingAssetRatiosBaseUnit: [
-          toBaseUnit(token0PoolRatio.toString(), pair.token0.decimals ?? 18),
-          toBaseUnit(token1PoolRatio.toString(), pair.token1.decimals ?? 18),
-        ] as const,
-        name: `${assets.byId[underlyingAssetIds[0]]?.symbol}/${
-          assets.byId[underlyingAssetIds[1]]?.symbol
-        } Pool`,
-        rewardAssetIds: [],
-        isClaimableRewards: false,
-      },
-    },
+    const assetId0 =
+      // Uniswap uses ERC20 WETH under the hood, but we want to display it as ETH
+      token0Address === WETH_TOKEN_CONTRACT_ADDRESS
+        ? ethAssetId
+        : toAssetId({
+            assetNamespace: 'erc20',
+            assetReference: token0Address,
+            chainId,
+          })
+    // Uniswap uses ERC20 WETH under the hood, but we want to display it as ETH
+    const assetId1 =
+      token1Address === WETH_TOKEN_CONTRACT_ADDRESS
+        ? ethAssetId
+        : toAssetId({
+            assetNamespace: 'erc20',
+            assetReference: token1Address,
+            chainId,
+          })
+    const underlyingAssetIds = [assetId0, assetId1] as const
+    const lpAsset = assets.byId[opportunityId]
+
+    if (bnOrZero(token0MarketData?.price).isZero()) {
+      continue
+    }
+
+    const token0Price = token0MarketData.price
+
+    const { assetReference } = fromAssetId(opportunityId)
+    // Checksum
+    const contractAddress = ethers.utils.getAddress(assetReference)
+    const uniV2LPContract = getOrCreateContractByType({
+      address: contractAddress,
+      type: ContractType.UniV2Pair,
+    })
+    const apy = bnOrZero(apr).div(100).toString()
 
+    // Getting the ratio of the LP token for each asset
+    const totalSupplyBaseUnit =
+      zapperAppBalanceData?.supply && lpAsset
+        ? bnOrZero(zapperAppBalanceData.supply).times(bn(10).pow(lpAsset.precision)).toString()
+        : (await uniV2LPContract.totalSupply()).toString()
+
+    const token0ReservesBaseUnit = bnOrZero(
+      bnOrZero(bnOrZero(token0Reserves.toString()).toString()),
+    ).times(bn(10).pow(lpAsset?.precision ?? 18))
+    const token1ReservesBaseUnit = bnOrZero(
+      bnOrZero(bnOrZero(token1Reserves?.toString()).toString()),
+    ).times(bn(10).pow(lpAsset?.precision ?? 18))
+    const token0PoolRatio = token0ReservesBaseUnit.div(totalSupplyBaseUnit).toString()
+    const token1PoolRatio = token1ReservesBaseUnit.div(totalSupplyBaseUnit).toString()
+    // Amount of token 0 in liquidity pool
+    const token0ReservesCryptoPrecision = bnOrZero(token0ReservesBaseUnit?.toString()).div(
+      bn(10).pow(token0Decimals ?? 18),
+    )
+
+    const totalLiquidityFiat = zapperAppBalanceData
+      ? bnOrZero(zapperAppBalanceData.dataProps.liquidity)
+      : token0ReservesCryptoPrecision.times(token0Price).times(2)
+    const tvl = totalLiquidityFiat.toString()
+    const price = bnOrZero(tvl)
+      .div(bnOrZero(totalSupplyBaseUnit.toString()).div(bn(10).pow(18)))
```

Comment:
this 18 because LP token are always 18?

Reply:
Correct yeah, that's the LP asset precision, 18 for all LPs - 50c355896e0fb90fbdbccb977de9350e5f3b4042